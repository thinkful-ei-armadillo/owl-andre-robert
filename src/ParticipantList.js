import React, { Component } from 'react';
//import './ParticipantList.css';
//import Participant from './Participant';

class ParticipantList extends Component {
  render() {
    return (
      <section className="participant-list">
        {/* {this.props.participants.map((p, index) => <Participant key={index} />)} */}
      </section>
    );
  }
}

export default ParticipantList;
