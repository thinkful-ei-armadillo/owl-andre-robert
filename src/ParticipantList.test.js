import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ParticipantList from './ParticipantList';
import {STORE} from './store';

describe('Testing participant list', () => {
  // smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ParticipantList></ParticipantList>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  //snapshot test
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<ParticipantList></ParticipantList>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});