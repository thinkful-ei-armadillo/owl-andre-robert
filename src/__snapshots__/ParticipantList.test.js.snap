// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing participant list renders the UI as expected 1`] = `
<section
  className="participant-list"
>
  <div
    className="participant"
  >
    <img
      alt="user avatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
    <div>
      <p
        className="msg-name"
      >
        Koren Templeton
      </p>
      <p
        className="session-indicator"
      >
        in session
      </p>
    </div>
  </div>
  <div
    className="participant"
  >
    <img
      alt="user avatar"
      src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
    />
    <div>
      <p
        className="msg-name"
      >
        Caty Flucker
      </p>
      <p
        className="session-indicator"
      >
        in session
      </p>
    </div>
  </div>
  <div
    className="participant"
  >
    <img
      alt="user avatar"
      src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
    />
    <div>
      <p
        className="msg-name"
      >
        Frank Runciman
      </p>
      <p
        className="session-indicator"
      >
        in session
      </p>
    </div>
  </div>
  <div
    className="participant"
  >
    <img
      alt="user avatar"
      src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
    />
    <div>
      <p
        className="msg-name"
      >
        Ashla Attwool
      </p>
      <p
        className="session-indicator"
      >
        in session
      </p>
    </div>
  </div>
  <div
    className="participant"
  >
    <img
      alt="user avatar"
      src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
    />
    <div>
      <p
        className="msg-name"
      >
        Axe Kubicka
      </p>
      <p
        className="disabled session-indicator"
      >
        not in session
      </p>
    </div>
  </div>
</section>
`;
